### ISSUE TITLE
[Broker] Better support in case of non-response of cloud providers

### ISSUE THREAD

**Issue by geonux** [CS-SI/SafeScale#41]
Quite regularly, the cloud provider is unable to deliver the desired machines on time.

- Improve error handling,
- make new attempts automatically,
- if more than one node is requested, allow access to the nodes until the other nodes are mounted,
- make the message returned to users understandable.

**Comment by oscarpicas** [comment]
### About error Handling:
Here is the output of broker and brokerd when a timeout is triggered:

**Broker terminal:** 
$ broker network create gorilla
`Error response from daemon : Failure waiting for gateway 'gw-gorilla' to finish provisioning and being accessible through SSH`

**Brokerd terminal:**
```
ERRO[2018-10-22 15:20:11] server '217.182.92.15' is not ready yet : retries timed out after 3s, Error reading user_data.log: exit status 255
Failure waiting for gateway 'gw-gorilla' to finish provisioning and being accessible through SSH
github.com/CS-SI/SafeScale/broker/daemon/services.(*NetworkService).Create
	/home/oscar/GoLand/src/github.com/CS-SI/SafeScale/broker/daemon/services/network.go:160
github.com/CS-SI/SafeScale/broker/daemon/commands.(*NetworkServiceServer).Create
	/home/oscar/GoLand/src/github.com/CS-SI/SafeScale/broker/daemon/commands/network.go:50
github.com/CS-SI/SafeScale/broker._NetworkService_Create_Handler
```

_What information would you like to see on "broker terminal" ?_

**Comment by geonux** [comment]
Les logs daemon :

```
./docker.safescale.run
root@a3f62637d847:/# brokerd
2018/10/18 18:31:47 Checking configuration
2018/10/18 18:31:47 Starting server
2018/10/18 18:31:47 Registering services
2018/10/18 18:31:47 Ready to serve :-)
INFO[2018-10-18 18:32:07] Tenant Set called 'fe'
INFO[2018-10-18 18:32:40] Current tenant is now 'fe'
INFO[2018-10-18 18:33:39] Tenant Set called 'fe'
INFO[2018-10-18 18:33:39] Tenant 'fe' is already selected
INFO[2018-10-18 18:33:39] Tenant Get called
INFO[2018-10-18 18:34:11] Tenant Get called
INFO[2018-10-18 18:34:11] Create Network called 'net-chronos'
INFO[2018-10-18 18:35:00] Waiting until gateway '' is finished  
provisioning and is available through SSH ...
INFO[2018-10-18 18:39:11] Ssh Host called 'chronos-master-1'
ERRO[2018-10-18 18:43:35] server '90.84.46.201' is not ready yet :  
retries timed out after 5m0s, Error reading user_data.log: bash: ssh:  
command not found
exit status 127
Failure waiting for gateway 'gw-net-chronos' to finish provisioning  
and being accessible through SSH
github.com/CS-SI/SafeScale/broker/daemon/services.(*NetworkService).Create
	/home/sdorgan/Desktop/Projects/go/src/github.com/CS-SI/SafeScale/broker/daemon/services/network.go:143
github.com/CS-SI/SafeScale/broker/daemon/commands.(*NetworkServiceServer).Create
	/home/sdorgan/Desktop/Projects/go/src/github.com/CS-SI/SafeScale/broker/daemon/commands/network.go:50
github.com/CS-SI/SafeScale/broker._NetworkService_Create_Handler
	/home/sdorgan/Desktop/Projects/go/src/github.com/CS-SI/SafeScale/broker/broker.pb.go:2453
github.com/CS-SI/SafeScale/vendor/google.golang.org/grpc.(*Server).processUnaryRPC
	/home/sdorgan/Desktop/Projects/go/src/github.com/CS-SI/SafeScale/vendor/google.golang.org/grpc/server.go:982
github.com/CS-SI/SafeScale/vendor/google.golang.org/grpc.(*Server).handleStream
	/home/sdorgan/Desktop/Projects/go/src/github.com/CS-SI/SafeScale/vendor/google.golang.org/grpc/server.go:1208
github.com/CS-SI/SafeScale/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1
	/home/sdorgan/Desktop/Projects/go/src/github.com/CS-SI/SafeScale/vendor/google.golang.org/grpc/server.go:686
runtime.goexit
	/usr/lib/go-1.10/src/runtime/asm_amd64.s:2361
INFO[2018-10-18 18:44:11] Ssh Host called 'chronos-node-1'
INFO[2018-10-18 18:44:11] Ssh Host called 'chronos-node-2'
INFO[2018-10-18 18:49:11] Create NAS called, name: chronos-nas
INFO[2018-10-18 18:54:11] Mount NAS called, name chronos-nas
INFO[2018-10-18 18:54:11] Mount NAS called, name chronos-nas
INFO[2018-10-18 18:54:11] Ssh Host called 'chronos-master-1'
ERRO[2018-10-18 19:03:30] <nil>
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xb67580]

goroutine 3089 [running]:
github.com/CS-SI/SafeScale/broker/utils.ToPBNas(...)
	/home/sdorgan/Desktop/Projects/go/src/github.com/CS-SI/SafeScale/broker/utils/converters.go:121
github.com/CS-SI/SafeScale/broker/daemon/commands.(*NasServiceServer).Create(0x138a7b0, 0xe7e600, 0xc4211da4b0, 0xc420222b40, 0x138a7b0, 0xc4211da3f0,  
0xc79b20)
	/home/sdorgan/Desktop/Projects/go/src/github.com/CS-SI/SafeScale/broker/daemon/commands/nas.go:56  
+0x390
github.com/CS-SI/SafeScale/broker._NasService_Create_Handler(0xcd44c0,  
0x138a7b0, 0xe7e600, 0xc4211da4b0, 0xc4202ada40, 0x0, 0x0, 0x0,  
0xe7e580, 0xc4200bc010)
	/home/sdorgan/Desktop/Projects/go/src/github.com/CS-SI/SafeScale/broker/broker.pb.go:3552  
+0x241
github.com/CS-SI/SafeScale/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc420001500, 0xe82760, 0xc4204f4300, 0xc4202aaa00, 0xc420394ae0,0x135ee00, 0x0, 0x0,  
0x0)
	/home/sdorgan/Desktop/Projects/go/src/github.com/CS-SI/SafeScale/vendor/google.golang.org/grpc/server.go:982  
+0x4f9
github.com/CS-SI/SafeScale/vendor/google.golang.org/grpc.(*Server).handleStream(0xc420001500, 0xe82760, 0xc4204f4300, 0xc4202aaa00,  
0x0)
	/home/sdorgan/Desktop/Projects/go/src/github.com/CS-SI/SafeScale/vendor/google.golang.org/grpc/server.go:1208  
+0x1318
github.com/CS-SI/SafeScale/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc420c40dd0, 0xc420001500, 0xe82760, 0xc4204f4300,  
0xc4202aaa00)
	/home/sdorgan/Desktop/Projects/go/src/github.com/CS-SI/SafeScale/vendor/google.golang.org/grpc/server.go:686  
+0x9f
created by  
github.com/CS-SI/SafeScale/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
	/home/sdorgan/Desktop/Projects/go/src/github.com/CS-SI/SafeScale/vendor/google.golang.org/grpc/server.go:684  
+0xa1

```

Les logs broker :

```
./chronos_deploy.sh fe
Tenant 'fe' set
2018/10/18 18:34:11 Creating infrastructure for cluster 'chronos'
2018/10/18 18:34:11 Creating Network 'net-chronos'
failed to create cluster: failed to create Network 'net-chronos': rpc  
error: code = DeadlineExceeded desc = context deadline exceeded
ssh connect took too long (> 5m0s) to respond
ssh connect took too long (> 5m0s) to respond
ssh connect took too long (> 5m0s) to respond
Error response from daemon: creation of nas took too long (> 5m0s) to  
respond (may eventually succeed)
Error response from daemon: Cannot mount NAS: Unable to find NAS 'chronos-nas'
Error response from daemon: Cannot mount NAS: Unable to find NAS 'chronos-nas'
ssh connect took too long (> 5m0s) to respond
```

**Comment by oscarpicas** [comment]
Commit cc58760a3adaf48a187fe696207f6deec3ae01f7 should fix your problem.
If something else happens, ***brokerd*** stores relevant log information in file "brokerd-session.log"

**Comment by vplanche-cssi** [comment]
SafeScale had made dramatic progress in regard of Cloud Provider failure tolerance. We advise you to have a look to the recently published v21.05.0 release candidates.
